vscode.sh

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/bash
+
+flatpak='flatpak-spawn --host flatpak'
+
+# In order for debugging to work, the the Unity VS Code extension needs to be able to find
+# Unity Editor's PID. Furthermore, it should be relatively unique to avoid conflicts.
+
+# The socket to connect to is found by taking the Unity PID modulo 100 and adding 56000:
+# https://github.com/Unity-Technologies/MonoDevelop.Debugger.Soft.Unity/blob/usedForVSCodeRelease/UnityProcessDiscovery.cs#L93
+
+# The "solution" is to grab a random number and use that as the target PID, then spawn a ton of
+# processes before our Unity fake shell script to ensure the PID is what we want.
+
+# 4 comes because bash will start as PID 2 inside the sandbox, and the cp will be PID
+#. Therfore, we need the minimum target PID to be 4.
+
+# An socat is then set up on the Unity end to connect the socket our PID exposes and the socket
+# that Unity Editor is on.
+
+target_pid=$((4 + RANDOM % 100))
+target_port=$((56000 + target_pid))
+unity_pid=$(pidof Unity)
+unity_port=$((56000 + unity_pid % 1000))
+
+read -r -d '' start_vscode <<'EOF'
+target_pid="$1"
+shift
+
+cp /usr/bin/sleep /tmp/Unity
+
+for (( i=4; i < $target_pid; i++ )); do /usr/bin/true; done
```

Comment:
My goodness... Given that we fetch the real Unity pid, is it actually worthwhile that the we loop like this up to 100 times and change the local fake Unity pid? Can't we just do something more like run fake Unity, check the corresponding socat port is available, if not, kill Unity and try again? So in the happy case (56004 is available) we will not loop at all.